src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/ApplicationInsightsTelemetryPublisher.cs

```diff
@@ -66,5 +71,24 @@ private void ValidateEvent(EvaluationEvent evaluationEvent)
                 throw new ArgumentNullException(nameof(evaluationEvent.FeatureDefinition));
             }
         }
+
+        private static string ToString(VariantAssignmentReason reason)
+        {
```

Comment:
Can you add `Debug.Assert(reason != VariantAssignmentReason.None);` at the beginning of the method. It helps add clarity to the reader that this is accounted for and is not missing.